base/plugins.cpp

```diff
@@ -256,6 +265,69 @@ void PluginManager::addPluginProvider(PluginProvider *pp) {
 	_providers.push_back(pp);
 }
 
+Plugin *PluginManager::giveEngineFromMetaEngine(const Plugin *plugin) {
+	assert(plugin->getType() == PLUGIN_TYPE_METAENGINE);
+
+	Plugin *enginePlugin = nullptr;
+	bool found = false;
+
+	// Use the engineID from MetaEngine for comparasion.
+	Common::String metaEnginePluginName = plugin->getEngineId();
+	PluginMan.loadFirstPlugin();
+	do {
+		PluginList pl = PluginMan.getPlugins(PLUGIN_TYPE_ENGINE);
+		// Iterate over all engine plugins.
+		for (PluginList::const_iterator itr = pl.begin(); itr != pl.end(); itr++) {
+			// The getName() provides a name which is similiar to getEngineId.
+			// Because engines are engines themselves, this function is simply named getName.
+			Common::String enginePluginName((*itr)->getName());
+
+			if (metaEnginePluginName.equalsIgnoreCase(enginePluginName)) {
+				enginePlugin = (*itr);
+				found = true;
+				break;
+			}
+		}
+	} while (!found && PluginMan.loadNextPlugin());
+
+	if (enginePlugin) {
+		warning("MetaEngine: %s \t matched to \t Engine: %s", plugin->getName(), enginePlugin->getFileName());
```

Comment:
Is that really a warning?

Reply:
Ah, sorry, I guess I didn't point it out. Throughout the tree, I have placed many warning messages. These just exist for review/test purposes, because it will be way easier to track down what's happening if anyone faces any issues.

I plan to remove all unnecessary comments, warning statements once the PR looks nice and near merge. So, I'd request you to ignore these warning statements, they're just there temporarily (most of them anyway).